types/src/transaction/authenticator.rs

```diff
@@ -17,6 +20,98 @@ use rand::{rngs::OsRng, Rng};
 use serde::{Deserialize, Serialize};
 use std::{convert::TryFrom, fmt, str::FromStr};
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum TransactionAuthenticator {
+    /// Single-agent transaction.
+    SingleAgent { sender: AccountAuthenticator },
```

Comment:
I have a preference (but not too strong) for keeping these two separate since we have stronger guarantees by the construction of the data with this approach. Also, if in the future we add more transaction authentication mechanisms I think separating them out like this makes sense. @emmazzz What are your thoughts?